contracts/contracts/strategies/BaseConvexMetaStrategy.sol

```diff
@@ -0,0 +1,180 @@
+     * InitializableAbstractStrategy initializer as Curve strategies don't fit
+     * well within that abstraction.
+     * @param _rewardTokenAddresses Address of CRV & CVX
+     * @param _assets Addresses of supported assets. MUST be passed in the same
+     *                order as returned by coins on the pool contract, i.e.
+     *                DAI, USDC, USDT
+     * @param _pTokens Platform Token corresponding addresses
+     * @param _initAddresses Various addresses containing the following:
+     *  - _platformAddress Address of the Curve 3pool
+     *  - _vaultAddress Address of the vault
+     *  - _cvxDepositorAddress Address of the Convex depositor(AKA booster) for this pool
+     *  - _metapoolAddress Address of the Curve MetaPool
+     *  - _metapoolMainToken Address of Main metapool token
+     *  - _cvxRewardStakerAddress Address of the CVX rewards staker
+     * @param _cvxDepositorPTokenId Pid of the pool referred to by Depositor and staker
+     */
+    function initialize(
+        address[] calldata _rewardTokenAddresses, // CRV + CVX
+        address[] calldata _assets,
+        address[] calldata _pTokens,
+        /**
+         * in the following order: _platformAddress(3Pool address), _vaultAddress, _cvxDepositorAddress
+         * _metapoolAddress, _metapoolMainToken, _cvxRewardStakerAddress, _metapoolLPToken
+         */
+        address[] calldata _initAddresses,
+        uint256 _cvxDepositorPTokenId
+    ) external onlyGovernor initializer {
+        require(_assets.length == 3, "Must have exactly three assets");
+        require(
+            _initAddresses.length == 7,
+            "_initAddresses must have exactly seven items"
+        );
+        // Should be set prior to abstract initialize call otherwise
+        // abstractSetPToken calls will fail
+        cvxDepositorAddress = _initAddresses[2];
+        pTokenAddress = _pTokens[0];
+        metapool = ICurveMetaPool(_initAddresses[3]);
+        metapoolMainToken = IERC20(_initAddresses[4]);
+        cvxRewardStakerAddress = _initAddresses[5];
+        metapoolLPToken = IERC20(_initAddresses[6]);
+        cvxDepositorPTokenId = _cvxDepositorPTokenId;
+
+        metapoolAssets = [metapool.coins(0), metapool.coins(1)];
+        super._initialize(
+            _initAddresses[0],
+            _initAddresses[1],
+            _rewardTokenAddresses,
+            _assets,
+            _pTokens
+        );
+        _approveBase();
+    }
+
+    /**
+     * @dev Get the total asset value held in the platform
+     * @param _asset      Address of the asset
+     * @return balance    Total value of the asset in the platform
+     */
+    function checkBalance(address _asset)
+        public
+        view
+        virtual
+        override
+        returns (uint256 balance)
+    {
+        require(assetToPToken[_asset] != address(0), "Unsupported asset");
+        balance = 0;
+        uint256 assetDecimals = Helpers.getDecimals(_asset);
```

Comment:
Super tiny: This isn't used until the very last line of the function. Would be clearer and less mental overhead to move it to the second last line.

Reply:
👍 moved